phoenix-core/src/it/java/org/apache/phoenix/end2end/OnDuplicateKeyIT.java

```diff
@@ -720,6 +747,470 @@ public void testMultiplePartialUpdatesInSameBatch() throws Exception {
+            assertEquals("abc", rs.getString("pk"));
+            assertEquals(1, rs.getInt("counter1"));
+            assertEquals(10, rs.getInt("counter2"));
+            assertEquals("NONE", rs.getString("approval"));
+            assertFalse(rs.next());
+
+            List<Long> timestampList1 = getAllColumnsLatestCellTimestamp(conn, tableName);
+            assertTrue(timestampList1.get(0) > timestampList0.get(0)
+                    && timestampList1.get(1) > timestampList0.get(1)
+                    && timestampList1.get(2) > timestampList0.get(2)
+                    && timestampList1.get(3) > timestampList0.get(3));
+
+            // Case 2: timestamps of counter2 and approval updated
+            actualReturnValue = conn.createStatement().executeUpdate(dml);
+            assertEquals(1, actualReturnValue);
+
+            rs = conn.createStatement().executeQuery(dql);
+            assertTrue(rs.next());
+            assertEquals("abc", rs.getString("pk"));
+            assertEquals(1, rs.getInt("counter1"));
+            assertEquals(11, rs.getInt("counter2"));
+            assertEquals("MANAGER_APPROVAL", rs.getString("approval"));
+            assertFalse(rs.next());
+
+            List<Long> timestampList2 = getAllColumnsLatestCellTimestamp(conn, tableName);
+            assertEquals(timestampList1.get(1), timestampList2.get(1)); // counter1 NOT updated
+            assertTrue(timestampList2.get(0) > timestampList1.get(0)
+                    && timestampList2.get(2) > timestampList1.get(2)
+                    && timestampList2.get(3) > timestampList1.get(3));
+
+            // Case 3: all timestamps NOT updated, including empty column
+            actualReturnValue = conn.createStatement().executeUpdate(dml);
+            assertEquals(0, actualReturnValue);
+
+            rs = conn.createStatement().executeQuery(dql);
+            assertTrue(rs.next());
+            assertEquals("abc", rs.getString("pk"));
+            assertEquals(1, rs.getInt("counter1"));
+            assertEquals(11, rs.getInt("counter2"));
+            assertEquals("MANAGER_APPROVAL", rs.getString("approval"));
+            assertFalse(rs.next());
+
+            List<Long> timestampList3 = getAllColumnsLatestCellTimestamp(conn, tableName);
+            assertEquals(timestampList2.get(0), timestampList3.get(0));
+            assertEquals(timestampList2.get(1), timestampList3.get(1));
+            assertEquals(timestampList2.get(2), timestampList3.get(2));
+            assertEquals(timestampList2.get(3), timestampList3.get(3));
+        }
+    }
+
+    @Test
+    public void testColumnsTimestampUpdateWithIntentionalUpdate() throws Exception {
+        Properties props = PropertiesUtil.deepCopy(TEST_PROPERTIES);
+        String tableName = generateUniqueName();
+        try (Connection conn = DriverManager.getConnection(getUrl(), props)) {
+            conn.setAutoCommit(true);
+
+            String ddl = "create table " + tableName +
```

Comment:
Can you also run this test by adding `COLUMN_ENCODED_BYTES = 0` to your create table ddl.